luatest/utils.lua

```diff
@@ -61,23 +61,33 @@ end
 
 -- Check if line of stack trace comes from inside luatest.
 local function is_luatest_internal_line(s)
-    return s:find('[/\\]luatest[/\\]') or s:find('bin[/\\]luatest')
+    return s:find('bin[/\\]luatest') or
+        (s:find('[/\\]luatest[/\\]') and
+         not s:find('[/\\]luatest[/\\]test[/\\]'))
 end
 
 function utils.strip_luatest_trace(trace)
     local lines = trace:split('\n')
-    local result = {lines[1]} -- always keep 1st line
-    local keep = true
-    for i = 2, table.maxn(lines) do
+    local result = {}
+    local keep = false
+    for i = table.maxn(lines), 2, -1 do
         local line = lines[i]
-        -- `[C]:` lines don't change context
-        if not line:find('^%s+%[C%]:') then
+        -- `[C]:`, `...`, `eval:` lines don't change context
+        if not (line:find('^%s+%[C%]:') or
+                line:find('^%s+%.%.%.$') or
+                line:find('^%s+eval:')) then
             keep = not is_luatest_internal_line(line)
         end
         if keep then
             table.insert(result, line)
         end
```

Comment:
The logic is not very clear here. As I understood the idea is to strip all the frames that originate from luatest and their corresponding `[C]` and similar frames.

However, we start stripping from the bottom all the frames (`keep = false`) without checking, whether they correspond to luatest. Why?

If my understanding above is correct, we want to keep non-luatest `[C]` and other frames. Shouldn't we check whether the line correspond to luatest first and then check for `[C]`?

At least, some clarification would be helpful.

Reply:
Actually, `keep` should be `true` initially, thanks for noticing. Fixed it. This doesn't change the output because a stack trace starts with `bin/luatest`.